forem/forem
# Update Android PN integration fixes

## What type of PR is this? (check all applicable)


## Description

This PR restructures the Android PN payload to allow for the app to handle them correctly.

We found a problem in the native integration code for Android which was causing an error that affected PN registration, the Podcast player & Video player, so the fix is included here.

## QA Instructions, Screenshots, Recordings

1. Boot web server locally and browse using the Android app
2. Podcast player should work as expected
3. After signing in/out the PN integration should work as expected
   - `/admin/apps/consumer_apps` should keep track of them as users sign in/out
4. Push Notifications with the proper payload should be sent

This is something we tested locally with @rt4914 so there's no need to go over the entire process if you don't something required for this (Android device, ngrok setup, etc.). Regardless, a LGTM would be greatly appreciated 😄 

### UI accessibility concerns?

None - Changes only impact the glue keeping Native <> web integration working

## Added/updated tests?

- [x] No, and this is why: Mocking functions here is quite tricky (mocking a JS function from a system/cypress spec). I'm a bit torn about this one because I do want to add some specs to avoid regressions. For now I'd be okay with merging this as is since it's a bug fix, but open to hearing the opinion of others on this.

## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: Part of the Android release project to be communicated by another department

## [optional] What gif best describes this PR or how it makes you feel?

![spongebob weird dance](https://media.giphy.com/media/9d1Fo0XyIYXzW/giphy.gif)
